test/data.js

```diff
@@ -0,0 +1,195 @@
+            for(let i = 0; i<stream.n_features; i++){
+                assert.equal(expected_names[i], stream.feature_names[i]);
+            }
+        });
+
+        let expected_targets = [0, 1, 2];
+        it('test expected targets', function(){
+            for(let i = 0; i<stream.n_classes; i++){
+                assert.equal(expected_targets[i], stream.target_values[i]);
+            }
+        });
+
+        it('test target names', function(){
+            assert.equal('target_0', stream.target_names[0]);
+        });
+
+        it('test if n_features equal 21', function(){
+            assert.equal(21, stream.n_features);
+        });
+
+        it('test n:cat_features', function(){
+            assert.equal(0, stream.n_cat_features);
+        });
+
+        it('test n_num_features', function(){
+            assert.equal(21, stream.n_num_features)
+        });
+
+        it('test n_targets', function(){
+            assert.equal(1, stream.n_targets)
+        })
+
+        let expected_info = 'Waveform Generator:'
+                            +'\n n_classes: 3'
+                            +'\n n_num_features: 21'
+                            +'\n n_cat_features: 0'
+                            +'\n has_noise: false'
+        it('test info', function(){
+            assert.equal(expected_info, stream.get_info());
+        });
+
+        it('test if stream has more samples', function(){
+            assert.equal(true, stream.has_more_samples());
+        });
+
+        it('test if stream is restartable', function(){
+            assert.equal(true, stream.is_restartable())
+        });
+
+        it('test next_sample()', function(){
+            next_sample = stream.next_sample();
+            assert.equal(21, next_sample[0][0].length);
+            assert.equal(1, next_sample[1].length);
+            assert.ok(next_sample[1] == 0 || next_sample[1] == 1 || next_sample[1] ==2);
+        });
+
+        it('test next_sample(batch_size = 2)', function(){
+            next_sample = stream.next_sample(2);
+            assert.equal(21, next_sample[0][0].length);
+            assert.equal(21, next_sample[0][1].length);
+            assert.equal(2, next_sample[1].length);
+            assert.ok(next_sample[1][0] == 0 || next_sample[1][0] == 1 || next_sample[1][0] == 2);
+            assert.ok(next_sample[1][1] == 0 || next_sample[1][1] == 1 || next_sample[1][1] == 2);
+        });
+
+        it('test last_sample()', function(){
+            next_sample = stream.next_sample(2);
+            last_sample = stream.last_sample();
+            for(let i = 0; i<stream.n_features; i++){
+                assert.equal(next_sample[0][0][i], last_sample[0][0][i]);
+                assert.equal(next_sample[0][1][i], last_sample[0][1][i]);
+            }
+            for(let i = 0; i<2; i++){
+                assert.equal(next_sample[1][i], last_sample[1][i]);
+                assert.equal(next_sample[1][i], last_sample[1][i]);
+            }
+        });
+
```

Comment:
Can we add a test to check the distribution of samples of the generator? Something to check whether it returns what it is expected also in terms of content, not just syntax?
`it('should return waveform distribution of the samples')`
Should histogram over a longer period return a uniform distribution? What is the period length?